src/tools/gravity-parseList.c

```diff
@@ -108,7 +113,7 @@ static inline bool __attribute__((pure)) valid_abp_domain(const char *line, cons
 			return false;
 
 		// First four characters must be "@@||"
-		if(line[0] != '@' || line[1] != '@' || line[2] != '|' || line[3] != '|')
+		if (memcmp(line, "@@||", 4) != 0)
```

Comment:
Is calling a function with all the related overhead really faster than doing a direct comparison in code that can be highly efficiently implemented by the compiler?